File: android/src/main/java/com/biso/capacitor/plugins/mlkit/barcode/scanner/BarcodeFormat.java
package com.biso.capacitor.plugins.mlkit.barcode.scanner;

import com.google.mlkit.vision.barcode.common.Barcode;

// suppress uppercase rule for enum naming (this matches js and swift code)
@SuppressWarnings("java:S115")
public enum BarcodeFormat {
  Code128(Barcode.FORMAT_CODE_128),
  Code39(Barcode.FORMAT_CODE_39),
  Code93(Barcode.FORMAT_CODE_93),
  CodaBar(Barcode.FORMAT_CODABAR),
  DataMatrix(Barcode.FORMAT_DATA_MATRIX),
  EAN13(Barcode.FORMAT_EAN_13),
  EAN8(Barcode.FORMAT_EAN_8),
  ITF(Barcode.FORMAT_ITF),
  QRCode(Barcode.FORMAT_QR_CODE),
  UPCA(Barcode.FORMAT_UPC_A),
  UPCE(Barcode.FORMAT_UPC_E),
  PDF417(Barcode.FORMAT_PDF417),
  Aztec(Barcode.FORMAT_AZTEC);

  private final int format;

  BarcodeFormat(int formatFlag) {
    this.format = formatFlag;
  }

  public int getAsInt() {
    return format;
  }

  public static BarcodeFormat getFromInt(int intValue) {
    for (BarcodeFormat value : BarcodeFormat.values()) {
      if(value.format == intValue) {
        return value;
      }
    }
    return null;
  }
}

File: ios/Plugin/BarcodeFormat.swift
import MLKitBarcodeScanning

public enum BarcodeFormat: String, CaseIterable {
    case CODE_128
    case CODE_39
    case CODE_93
    case CODA_BAR
    case DATA_MATRIXt
    case EAN_13
    case EAN_8
    case ITF
    case QR_CODE
    case UPC_A
    case UPC_E
    case DF417
    case AZTEC
    
    var asInt: Int {
        switch self {
        case BarcodeFormat.CODE_128: return MLKitBarcodeScanning.BarcodeFormat.code128.rawValue
        case BarcodeFormat.CODE_39: return MLKitBarcodeScanning.BarcodeFormat.code39.rawValue
        case BarcodeFormat.CODE_93: return MLKitBarcodeScanning.BarcodeFormat.code93.rawValue
        case BarcodeFormat.CODA_BAR: return MLKitBarcodeScanning.BarcodeFormat.codaBar.rawValue
        case BarcodeFormat.DATA_MATRIXt: return MLKitBarcodeScanning.BarcodeFormat.dataMatrix.rawValue
        case BarcodeFormat.EAN_13: return MLKitBarcodeScanning.BarcodeFormat.EAN13.rawValue
        case BarcodeFormat.EAN_8: return MLKitBarcodeScanning.BarcodeFormat.EAN8.rawValue
        case BarcodeFormat.ITF: return MLKitBarcodeScanning.BarcodeFormat.ITF.rawValue
        case BarcodeFormat.QR_CODE: return MLKitBarcodeScanning.BarcodeFormat.qrCode.rawValue
        case BarcodeFormat.UPC_A: return MLKitBarcodeScanning.BarcodeFormat.UPCA.rawValue
        case BarcodeFormat.UPC_E: return MLKitBarcodeScanning.BarcodeFormat.UPCE.rawValue
        case BarcodeFormat.DF417: return MLKitBarcodeScanning.BarcodeFormat.PDF417.rawValue
        case BarcodeFormat.AZTEC: return MLKitBarcodeScanning.BarcodeFormat.aztec.rawValue
        }
    }
}
